src/swap/SwapReviewScreen.tsx

```diff
@@ -0,0 +1,374 @@
+import { Screens } from 'src/navigator/Screens'
+import { StackParamList } from 'src/navigator/types'
+import DisconnectBanner from 'src/shared/DisconnectBanner'
+import colors from 'src/styles/colors'
+import fontStyles from 'src/styles/fonts'
+import { Spacing } from 'src/styles/styles'
+import variables from 'src/styles/variables'
+import { Field } from 'src/swap/useSwapQuote'
+import { coreTokensSelector } from 'src/tokens/selectors'
+import { divideByWei, multiplyByWei } from 'src/utils/formatting'
+import Logger from 'src/utils/Logger'
+import networkConfig from 'src/web3/networkConfig'
+import { walletAddressSelector } from 'src/web3/selectors'
+
+type Props = StackScreenProps<StackParamList, Screens.SwapReviewScreen>
+
+const TAG = 'SWAP_REVIEW_SCREEN'
+
+export function SwapReviewScreen(props: Props) {
+  const { toToken, fromToken, swapAmount, updatedField } = props.route.params
+  const [loading, setLoading] = useState(false)
+  const [fetchSwapQuoteError, setFetchSwapQuoteError] = useState(false)
+  const [estimatedModalVisible, setEstimatedDialogVisible] = useState(false)
+  const [swapFeeModalVisible, setSwapFeeModalVisible] = useState(false)
+  const [swapInfo, setSwapInfo] = useState(null as any)
+  const coreTokens = useSelector(coreTokensSelector)
+  const walletAddress = useSelector(walletAddressSelector)
+
+  // Items set for remote config
+  const maxSlippagePercent = useSelector(maxSwapSlippagePercentageSelector)
+  const swapFeeEnabled = useSelector(swapFeeEnabledSelector)
+  const swapFeePercentage = useSelector(swapFeePercentageSelector)
+
+  // Remote configs converted to decimals strings
+  const maxSlippageDecimal = `${maxSlippagePercent / 100}`
+  const swapFeeDecimal = `${swapFeePercentage / 100}`
+
+  const { t } = useTranslation()
+  const dispatch = useDispatch()
+
+  // Token Symbols
+  const toTokenSymbol = coreTokens.find((token) => token.address === toToken)?.symbol
+  const fromTokenSymbol = coreTokens.find((token) => token.address === fromToken)?.symbol
+
+  useEffect(() => {
+    if (fetchSwapQuoteError) {
+      dispatch(showError(ErrorMessages.FETCH_SWAP_QUOTE_FAILED))
+    }
+  }, [fetchSwapQuoteError])
+
+  useEffect(() => {
+    ValoraAnalytics.track(SwapEvents.swap_review_screen_open, {
+      toToken,
+      fromToken,
+      buyAmount: swapAmount[Field.TO] ?? '',
+    })
+  }, [])
+
+  const getSwapInfo = async () => {
+    try {
+      setFetchSwapQuoteError(false)
+      setLoading(true)
+      const swapAmountInWei = multiplyByWei(swapAmount[updatedField]!)
+      const swapAmountParam = updatedField === Field.FROM ? 'sellAmount' : 'buyAmount'
+      const params = {
+        buyToken: toToken,
+        sellToken: fromToken,
+        [swapAmountParam]: swapAmountInWei.toString().split('.')[0],
+        slippagePercentage: maxSlippageDecimal,
+        userAddress: walletAddress,
+      }
+      const queryParams = new URLSearchParams({ ...(params as Record<string, string>) }).toString()
```

Comment:
same comment as in the hook, we shouldn't need the type cast for `params` if we handle the null `walletAddress` case :)